saleor/plugins/openid_connect/plugin.py

```diff
@@ -282,15 +287,19 @@ def external_refresh(
                 }
             )
         try:
-            user_permissions = None
-            if self.config.use_scope_permissions:
-                user_permissions = get_saleor_permissions_from_scope(
-                    token_data.get("scope")
-                )
             parsed_id_token = get_parsed_id_token(
                 token_data, self.config.json_web_key_set_url
             )
             user = get_user_from_token(parsed_id_token)
+
+            user_permissions = None
+            if self.config.use_scope_permissions:
+                scope = token_data.get("scope")
+                if scope:
+                    permissions = get_saleor_permissions_qs_from_scope(scope)
+                    user_permissions = get_saleor_permission_names(permissions)
+                    user.effective_permissions = permissions
```

Comment:
This part appeared two times in this file, maybe we can create a method from that?